registry/strings/helpers.go

```diff
@@ -272,3 +282,50 @@ func (sr *StringsRegistry) wordWrap(wrapLength int, newLineCharacter string, wra
 
 	return resultBuilder.String()
 }
+
+// swapFirstLetter swaps the first letter of the string 'str' to uppercase or
+// lowercase. The casing is determined by the 'casing' parameter.
+//
+// Parameters:
+//
+//	str string - the string to modify.
+//	casing casingType - the casing to apply to the first letter.
+//
+// Returns:
+//
+//	string - the modified string with the first letter in the desired casing.
+//
+// Example:
+//
+//	result := sr.swapFirstLetter("123hello", cassingUpper)
+//	fmt.Println(result) // Output: "123Hello"
+func swapFirstLetter(str string, casing casingType) string {
+	var conditionFunc func(r rune) bool
+	var updateFunc func(r rune) rune
+
+	if casing == cassingUpper {
+		conditionFunc = unicode.IsUpper
+		updateFunc = unicode.ToUpper
+	} else {
+		conditionFunc = unicode.IsLower
+		updateFunc = unicode.ToLower
+	}
```

Comment:
You defined a iota, OK you only have two values

You could have used a boolean value, but you didn't

So I think you should consider a switch here

```suggestion
	var conditionFunc func(r rune) bool
	var updateFunc func(r rune) rune

	switch casing {
	case cassingUpper:
		conditionFunc = unicode.IsUpper
		updateFunc = unicode.ToUpper
	case cassingLower:
		conditionFunc = unicode.IsLower
		updateFunc = unicode.ToLower
	default:
		panic("unsupported casingType")
	}
```

a panic is not elegant, but adding an error, make no sens while nothing else use it.

But maybe I'm nitpicking, maybe you will have a better idea than mine, now I raised this

Reply:
Wow I'm in another planet when I fix that, I don't follow my own rules 😮‍💨

no panic in the project: ❌ 
memory efficiency: ❌ 

Let me fix my work and never talk about that again ⚰️ 😂